Simulator/Recorders/Hdf5Recorder.cpp

```diff
@@ -169,20 +171,20 @@ void Hdf5Recorder::initDataSet()
    }
 
     // allocate and initialize data memories
-    burstinessHist_ = new int[static_cast<int>(Simulator::getInstance().getEpochDuration())];
-   spikesHistory_ = new int[static_cast<int>(Simulator::getInstance().getEpochDuration() * 100)];
-    memset(burstinessHist_, 0, static_cast<int>(Simulator::getInstance().getEpochDuration() * sizeof(int)));
-    memset(spikesHistory_, 0, static_cast<int>(Simulator::getInstance().getEpochDuration() * 100 * sizeof(int)));
+    burstinessHist_ = new int[static_cast<int>(simulator.getEpochDuration() * simulator.getNumEpochs())];
+    spikesHistory_ = new int[static_cast<int>(simulator.getEpochDuration() * simulator.getNumEpochs() * 100)];
+    memset(burstinessHist_, 0, static_cast<int>(simulator.getEpochDuration() * simulator.getNumEpochs() * sizeof(int)));
+    memset(spikesHistory_, 0, static_cast<int>(simulator.getEpochDuration() * simulator.getNumEpochs() * 100 * sizeof(int)));
```

Comment:
I assume this is the place where the epoch data buffers were reallocated to be much larger. If that isn't the case in BrainGrid, then it must be the case that there is code in BG that avoids the problem we're seeing in Graphitti. Rather than hack this solution, we should find where the missing pieces are in Graphitti and fix them. Otherwise, not only do these become enormous for long simulations, but there will be incompatibilities with the GPU code.

Reply:
Given fixes in following commits, looks good.